src/site/_includes/components/component-preview/__tests__/index.test.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint no-console: ["error", { allow: ["warn"] }] */
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { initComponentPreviewsIframe } from '..'
+
+const html = `
+  <div class="original-class js-guidance-component-preview-iframe" data-example-name="test-component" data-iframe-id="test-component--iframe" data-iframe-title="test-component-title">
+      <div class="original-class-content" >
+        Template content
+      </div>
+    </div>
+  </div>
+`
+
+describe('initComponentPreviewsIframe', () => {
+  const originalWarn = console.warn
+
+  beforeEach(() => {
+    console.warn = vi.fn()
```

Comment:
Maybe worth adding a comment about why this is here (to suppress the `[iFrameSizer][Host page: test-component--iframe] [init] IFrame(test-component--iframe) not found` warning I assume).